File: Crossbar/BarrierLine.py
class BarrierLine(object):
    def __init__(self, value = 0):
        self._value = value

    def get_value(self):
        return self._value

    def set_value(self, value):
        self._value = value

    def toggle(self):
        self._value = 1 - self._value

    def is_up(self):
        return not(self.is_down())

    def is_down(self):
        return bool(self._value)

File: crossbar/control-lines/BarrierLine.cpp
#include "BarrierLine.h"

BarrierLine::BarrierLine(double value) {
	this->value = value;
}

double BarrierLine::get_value() const {
	return this->value;
}

void BarrierLine::set_value(double value) {
	this->value = value;
}

bool BarrierLine::is_up() const {
	return (this->get_value() == 0);
}

bool BarrierLine::is_down() const {
	return (this->get_value() == 1);
}

bool BarrierLine::is_open() const {
	return this->is_down();
}

bool BarrierLine::is_closed() const {
	return this->is_up();
}

void BarrierLine::open() {
	this->set_value(1);
}

void BarrierLine::close() {
	this->set_value(0);
}

void BarrierLine::toggle() {
	this->set_value(1 - this->get_value());
}
